posthog/api/test/batch_exports/test_create.py

```diff
@@ -660,5 +694,135 @@ def test_cannot_create_a_batch_export_for_databricks_if_not_enabled(
+    assert schedule.schedule.spec.intervals[0].every == dt.timedelta(hours=1)
+    assert isinstance(schedule.schedule.action, ScheduleActionStartWorkflow)
+    assert schedule.schedule.action.workflow == "databricks-export"
+
+
+def test_creating_databricks_batch_export_fails_if_feature_flag_is_not_enabled(
+    client: HttpClient, temporal, organization, team, user, databricks_integration
+):
+    """Test that creating a Databricks batch export fails if the feature flag is not enabled."""
+
+    destination_data = {
+        "type": "Databricks",
+        "config": {
+            "http_path": "my-http-path",
+            "catalog": "my-catalog",
+            "schema": "my-schema",
+            "table_name": "my-table-name",
+            "integration_id": databricks_integration.id,
+        },
+    }
+
+    batch_export_data = {
+        "name": "my-databricks-destination",
+        "destination": destination_data,
+        "interval": "hour",
+    }
+
+    client.force_login(user)
+
+    response = create_batch_export(
+        client,
+        team.pk,
+        batch_export_data,
+    )
+
+    assert response.status_code == status.HTTP_403_FORBIDDEN, response.json()
+    assert "The Databricks destination is not enabled for this team." in response.json()["detail"]
+
+
+def test_creating_databricks_batch_export_fails_if_integration_is_missing(
+    client: HttpClient, temporal, organization, team, user, enable_databricks
+):
+    """Test that creating a Databricks batch export fails if the integration is missing.
+
+    Using integrations is the preferred way to handle credentials for batch exports going forward.
+    """
+
+    destination_data = {
+        "type": "Databricks",
+        "config": {
+            "http_path": "my-http-path",
+            "catalog": "my-catalog",
+            "schema": "my-schema",
+            "table_name": "my-table-name",
+            "integration_id": None,
+        },
+    }
+
+    batch_export_data = {
+        "name": "my-databricks-destination",
+        "destination": destination_data,
+        "interval": "hour",
+    }
+
+    client.force_login(user)
+    response = create_batch_export(
+        client,
+        team.pk,
+        batch_export_data,
+    )
+    assert response.status_code == status.HTTP_400_BAD_REQUEST, response.json()
+
+    assert response.json() == {
+        "type": "validation_error",
+        "code": "invalid_input",
+        "detail": "integration_id is required for Databricks batch exports",
+        "attr": "destination",
+    }
+
+
+def test_creating_databricks_batch_export_fails_if_integration_is_invalid(
+    client: HttpClient, temporal, organization, team, user, enable_databricks
+):
+    """Test that creating a Databricks batch export fails if the integration is invalid.
+
+    Using integrations is the preferred way to handle credentials for batch exports going forward.
+
+    In this case, the integration is missing the client_secret.
+    """
```

Comment:
question: Does integrations (our ourselves) also handle wrong types? Not sure if I have seen that yet...

Reply:
I've added some tests for the API now to ensure it's not possible to create an invalid integration in the first place